src/System.IO.FileSystem/src/System/IO/CharSpanExtensions.Windows.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+
+namespace System.IO
+{
+    internal static partial class CharSpanExtensions
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static unsafe int CompareOrdinal(ReadOnlySpan<char> first, ReadOnlySpan<char> second, bool ignoreCase = false)
+        {
+            int result = Interop.Kernel32.CompareStringOrdinal(
```

Comment:
why not start with
```c#
+            if (first.Length != second.Length)
 +                return false;
 +
 +            if (!ignoreCase)
 +                return first.SequenceEqual(second);
```
save calling into Windows to just do pointer comparisons.

Reply:
Oops, this comment was for EqualsOrdinal below.

Reply:
Fair enough